tests/e2e/main.py

```diff
@@ -0,0 +1,124 @@
+    """
+    url = url_prefix + '/logs/search?' + 'from=now-5m&limit=1000&query=aws.service%3D%22lambda%22%20AND%20content%3D%22' + message_content + '%22%20AND%20timestamp%3D%22' + str(epoch_timestamp_in_ms) + '%22&sort=-timestamp'
+    response = requests.get(url,
+                            headers=request_headers)
+
+    # check, if response status code is OK
+    if response.status_code != requests.codes.ok:
+        response.raise_for_status()
+
+    return response.json()
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--url-prefix',
+                        '-u',
+                        action='store',
+                        type=str,
+                        default='https://jxw01498.dev.dynatracelabs.com/api/v2',
+                        help='specify Dynatrace SaaS URL or Dynatrace AG URL')
+    parser.add_argument('--api-token',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify Dynatrace API Token')
+    parser.add_argument('--log-group-name',
+                        '-l',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify AWS CloudWatch Logs log group name')
+    parser.add_argument('--log-stream-name',
+                        '-s',
+                        action='store',
+                        type=str,
+                        default='',
+                        required=True,
+                        help='specify AWS CloudWatch Logs log stream name for a given log group')
+    parser.add_argument('--unique-message-content',
+                        '-m',
+                        action='store',
+                        type=str,
+                        default='Dynatrace-AWS-Log-Forwarder-End-to-End-test-message',
+                        help='specify a unique log event message content')
+    args = parser.parse_args()
+
+    epoch_timestamp_in_ms = int(time.time() * 1000)
+    request_headers       = {'Authorization': 'Api-Token ' + args.api_token,
+                            'Content-Type': 'application/json',
+                            'charset': 'utf-8'}
+    message_content       = 'Dynatrace-AWS-Log-Forwarder-End-to-End-test-' + args.unique_message_content
+
+    # Generate a test log event
+    generate_log_event(args.log_group_name,
+                       args.log_stream_name)
+
+    # Search Dynatrace logs for records with specific message content and timestamp
+    dynatrace_log_records_search_results = search_dynatrace_log_records(args.url_prefix,message_content)['results']
+    time_elapsed = 0
+    while bool(dynatrace_log_records_search_results) is False:
+        print('Waiting for log events to be picked up by the cluster...')
+        time.sleep(10)
```

Comment:
It's a bit different approach than described in the story:
```
use retry behavior to start verification early but if delay is bigger then expected then retry few times with e.g. 1 minute wait; e.g. initial wait 5 minutes and then 5 retries after 1 minute until test succeeds
```
Are we okay with that?

Reply:
I think it is fine, do you see any reason against?

Reply:
Nah, I'm good. Just wanted to attract your attention to it because maybe someone else would have a strong opinion. Let's keep it as is then

Reply:
I've tested this in several iterations and, if the forwarder was deployed properly, the delay was never bigger than 2min. However, this value could be subject to change when we start using the dedicated AG solution.